src/jsonpath.ts

```diff
@@ -0,0 +1,197 @@
+import { splitSDJWT } from './common';
+import { SD_DIGEST, SD_LIST_PREFIX } from './constants';
+import { combineSDJWT, decodeDisclosure, decodeJWT, isObject } from './helpers';
+import { Disclosure, GetHasher, Hasher } from './types';
+
+export type DigestMap = Map<string, Disclosure>;
+
+type getDisclosuresOptions = {
+  ignoreInvalid?: boolean;
+};
+
+/**
+ * List and Retrieve Disclosures using
+ * Explicit dot-notation JSONpath expression
+ */
+export class SdJwtDisclosureParser {
```

Comment:
If I understand everything correctly, it is a tree data structure. Maybe the name could reflect it somehow? `SdJwtDisclosureTree`, `SdJwtDisclosureMap`, `SdJwtDisclosureStruct`?
I'm just thinking out loud here.